### DISCUSSION TITLE
Make whole row clickable

### DISCUSSION THREAD

**Discussion by kyusungpark** [TanStack/table#5733] [Q&A]
Hi, I would like to add a behavior to rows where users would be navigated to a different page when they click anywhere on the role (except one where I have a dropdown component). How would I implement such behavior?

**Comment by KevinVandy** [comment]
Ad an onClick

**Reply by kyusungpark** [reply]
Is there like a easy to do it just like how we create columns instead of adding it on the tableRow?

**Reply by KevinVandy** [reply]
Making rows clickable is as easy as adding an onClick handler to them. Use proper `event stopPropogation()` in any cell onClicks to not accidentally pick up extra row clicks.

Guess I don't know what you're having trouble with?

**Reply by kyusungpark** [reply]
Okay, thank you, I was wondering if there is another way like when we create columns maybe something special for the whole row, but I guess no. 

```
{  accessorKey: key,
  enableSorting: isSortable,
  header: ({ column }) => (isSortable ? <SortableHeader label={label} column={column} /> : label),
  cell: ({ row }) => <DateCell date={row.original[key] as string} />, }
```

Thank you for the help!
